auth/auth-core/src/main/java/com/bazaarvoice/emodb/auth/shiro/PrincipalWithRoles.java

```diff
@@ -49,7 +50,7 @@ public UserIdentity toUserIdentity() {
 
             _userIdentity = new DefaultUserIdentity(
                     new javax.security.auth.Subject(true, ImmutableSet.of(this), ImmutableSet.of(), ImmutableSet.of()),
-                    this, roles);
+                    new BasicUserPrincipal(getId()), roles);
```

Comment:
Note:  This is the one change for printing the API key ID instead of the secret key to the request logs.